### ISSUE TITLE
device_type: cisco_ios_serial error using USB mini-Type B

### ISSUE THREAD

**Issue by Tes3awy** [ktbyers/netmiko#2288]
Hello @ktbyers 

In some models of Cisco devices, you can find a [USB mini-Type B console port](https://www.cisco.com/c/en/us/td/docs/switches/lan/catalyst9300/hardware/install/b_c9300_hig/b_c9300_hig_chapter_01001.html#concept_40CDA463956B43ACA9FFBBDAD6D3DA49__fig_fg4_nwt_41b) on the front of the device which lets you connect serially to that device. When I am connected to a Cisco device using that Type B console cable and trying to initiate a serial connection instance, I get an error. But when I use a standard Cisco console cable **(RJ-45)**, it works like a charm. I don't know what is the problem with the Type B cable? 

I know the two cables work with different technologies. But what is the issue in this case?

**The error I get:**
```bash
$ python serial_conn.py
Timed-out reading channel, pattern not found in output: terminal\ length\ 0

Router prompt not found: 'switch_'
```
**Code sample:**
```python
#!usr/bin/env python3

from netmiko import ConnectHandler as CH
from pprint import pprint

# Define a device
device = {
    "device_type": "cisco_ios_serial",
    "ip": "10.1.1.1",
    "username": "cisco",
    "password": "cisco",
    "serial_settings": {
        "port": "COM4",
    },
}

# Initiate a serial connection instance to device
with CH(**device) as net_connect:
    output = net_connect.send_command("show version", use_textfsm=True)
pprint(output)
```

**Comment by ktbyers** [comment]
Do you have it working outside of Netmiko (i.e. using some sort of terminal emulator on your computer)?

**Comment by Tes3awy** [comment]
> Do you have it working outside of Netmiko (i.e. using some sort of terminal emulator on your computer)?

Yes it works. I usually use the USB Type B console port because it's on the front of the device. The RJ-45 port is usually on the rear side of the device and sometimes difficult to reach.

USB Type B works with PuTTY, SecureCRT and MobaXTerm emulators.

**Comment by ktbyers** [comment]
What is the error you get when it fails. Can you post the full stack trace here?

**Comment by Tes3awy** [comment]
> 
> 
> What is the error you get when it fails. Can you post the full stack trace here?

```
Traceback (most recent call last):
  File "D:\Folder1\serial_conn.py", line 10, in <module>
    with ConnectHandler(**device) as net_connect:
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\site-packages\netmiko\ssh_dispatcher.py", line 326, in ConnectHandler
    return ConnectionClass(*args, **kwargs)
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\site-packages\netmiko\cisco\cisco_ios.py", line 17, in __init__      
    return super().__init__(*args, **kwargs)
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\site-packages\netmiko\base_connection.py", line 350, in __init__     
    self._open()
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\site-packages\netmiko\base_connection.py", line 356, in _open
    self._try_session_preparation()
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\site-packages\netmiko\base_connection.py", line 792, in _try_session_preparation
    self.session_preparation()
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\site-packages\netmiko\cisco\cisco_ios.py", line 24, in session_preparation
    self.set_base_prompt()
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\site-packages\tenacity\__init__.py", line 339, in wrapped_f
    return self(f, *args, **kw)
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\site-packages\tenacity\__init__.py", line 430, in __call__
    do = self.iter(retry_state=retry_state)
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\site-packages\tenacity\__init__.py", line 378, in iter
    raise retry_exc.reraise()
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\site-packages\tenacity\__init__.py", line 206, in reraise
    raise self.last_attempt.result()
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\concurrent\futures\_base.py", line 438, in result
    return self.__get_result()
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\concurrent\futures\_base.py", line 390, in __get_result
    raise self._exception
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\site-packages\tenacity\__init__.py", line 433, in __call__
    result = fn(*args, **kwargs)
  File "C:\Users\Username\AppData\Local\Programs\Python\Python39\lib\site-packages\netmiko\base_connection.py", line 1166, in set_base_prompt
    raise ValueError(f"Router prompt not found: {repr(prompt)}")
ValueError: Router prompt not found: 'switch_'
```

**Comment by ktbyers** [comment]
Does it work if you try this?

```python
# Define a device
device = {
    "device_type": "cisco_ios_serial",
    "ip": "10.1.1.1",
    "username": "cisco",
    "password": "cisco",
    "serial_settings": {
        "port": "COM4",
    },
   "fast_cli": False,
   "session_log": "my_output.txt",
}
```

Also do you see anything in the session_log (will be named "my_output.txt" given the device definition that I posted above).

**Comment by Tes3awy** [comment]
I tried using `send_command_timing("show version", use_textfsm=True)` instead and it worked. But there is something I noticed which is weird to me. I had to explicitly set `"fast_cli": False`!!. I can see in the `BaseConnection` class that `fast_cli` has `False` as its default value. _(I am using Netmiko v3.4.0)_

![Screenshot 2021-06-29 195832](https://user-images.githubusercontent.com/11368432/123845406-70349680-d914-11eb-834b-bb66842cbbbf.png)

**Comment by ktbyers** [comment]
`cisco_ios` overwrites the default and sets it to True (in its class).

https://github.com/ktbyers/netmiko/blob/v3.4.0/netmiko/cisco/cisco_ios.py#L15

Note, I just changed some of this behavior in the Netmiko `develop` branch, but this is True for Netmiko version 3.4.0 which is what is released on pypy (and version 3.4.0 is also what I linked to above).
